diff --git a/hif/pcie/pcie.c b/hif/pcie/pcie.c
index 1ce3213..526cb37 100644
--- a/hif/pcie/pcie.c
+++ b/hif/pcie/pcie.c
@@ -736,7 +736,8 @@ static struct device_node *pcie_get_device_node(struct ieee80211_hw *hw)
 	struct device_node *dev_node;
 
 	dev_node = pci_bus_to_OF_node(pcie_priv->pdev->bus);
-	wiphy_info(priv->hw->wiphy, "device node: %s\n", dev_node->full_name);
+	if (dev_node)
+		wiphy_info(priv->hw->wiphy, "device node: %s\n", dev_node->full_name);
 
 	return dev_node;
 }