Index: mwlwifi-2023-11-09-47b0cd8a/hif/pcie/pcie.c
===================================================================
--- mwlwifi-2023-11-09-47b0cd8a.orig/hif/pcie/pcie.c
+++ mwlwifi-2023-11-09-47b0cd8a/hif/pcie/pcie.c
@@ -685,7 +685,8 @@ static struct device_node *pcie_get_devi
 	struct device_node *dev_node;
 
 	dev_node = pci_bus_to_OF_node(pcie_priv->pdev->bus);
-	wiphy_info(priv->hw->wiphy, "device node: %s\n", dev_node->full_name);
+	if (dev_node)
+		wiphy_info(priv->hw->wiphy, "device node: %s\n", dev_node->full_name);
 
 	return dev_node;
 }
